src/util/getRandomValues.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+/*
+ * API shim for `window.crypto.getRandomValues` in the browser, for
+ * compatibility with browser, Node, Jest, and Observable.
+ *
+ * Forked from NPM `get-random-bytes` by Kenan Yildirim, which is
+ * released under the MIT License.
+ */
+
+/**
+ * Fill the given buffer with cryptographically secure random bytes. The
+ * buffer length must not exceed 65536.
+ */
+function getRandomValues(buf: Uint8Array): Uint8Array {
+  if (typeof window !== "undefined") {
+    if (window.crypto && window.crypto.getRandomValues) {
+      return window.crypto.getRandomValues(buf);
+    }
+    if (
+      typeof window.msCrypto === "object" &&
+      typeof window.msCrypto.getRandomValues === "function"
+    ) {
+      return window.msCrypto.getRandomValues(buf);
+    }
+  }
+
+  if (typeof require !== "undefined") {
+    // Late-import `crypto` to avoid `require` in Observable notebooks,
+    // and avoid using a literal `require(...)` to prevent Webpack from
+    // rewriting this. (Getting Webpack externals to work properly is .)
```

Comment:
seems like you had a thought to finish here

Reply:
Ah, yes; thanks. Let’s go with *daunting*.